autoscaler/controllers/common/processors.go

```diff
@@ -70,3 +78,90 @@ func GetGenericBatchProcessor() odigosv1.Processor {
 		},
 	}
 }
+
+type MatchCondition struct {
+	Name      string `mapstructure:"name"`
+	Namespace string `mapstructure:"namespace"`
+	Kind      string `mapstructure:"kind"`
+}
+
+func GenerateRoutingProcessors(
+	ctx context.Context,
+	kubeClient client.Client,
+	dests *odigosv1.DestinationList,
+) map[string]config.GenericMap {
+	logger := log.FromContext(ctx)
+	routingProcessors := make(map[string]config.GenericMap)
+
+	for _, dest := range dests.Items {
+
+		if dest.Spec.SourceSelector == nil {
+			continue
+		}
+
+		matchConditions := []string{}
+		if len(dest.Spec.SourceSelector.Namespaces) > 0 {
+			for _, namespace := range dest.Spec.SourceSelector.Namespaces {
+				matchConditions = append(matchConditions, fmt.Sprintf("%s/*/*", namespace))
+			}
+		}
+		if len(dest.Spec.SourceSelector.Groups) > 0 {
+			matchedSources := fetchSourcesByGroups(ctx, kubeClient, dest.Spec.SourceSelector.Groups, logger)
+			for _, source := range matchedSources {
+				key := fmt.Sprintf("%s/%s/%s", source.Spec.Workload.Namespace, source.Spec.Workload.Name, source.Spec.Workload.Kind)
```

Comment:
we usually write kind before name. e.g. `deployment-foo` and not `foo-deployment`
```suggestion
				key := fmt.Sprintf("%s/%s/%s", source.Spec.Workload.Namespace, source.Spec.Workload.Kind, source.Spec.Workload.Name)
```